lib/modules/manager/gradle/extract/catalog.ts

```diff
@@ -14,6 +15,28 @@ import type {
   VersionPointer,
 } from '../types';
 
+function findVersionIndex(
+  content: string,
+  depName: string,
+  version: string
+): number {
+  const contents = content.replace(regEx(/\r\n/g), '\n');
+  const eDn = escapeRegExp(depName);
+  const eVer = escapeRegExp(version);
+  const re = regEx(
+    `(id\\s*=\\s*)?['"]?${eDn}["']?((\\s*=\\s*)|:|,\\s*)(.*version(\\.ref)?(\\s*\\=\\s*))?["']?${eVer}['"]?`
```

Comment:
Are you sure single quotes are allowed in toml files?

Reply:
Please, prefer non-capturing groups when possible, i.e. `(?:foobar)` instead of `(foobar)`.

Reply:
michael, yes i tested single quotes, double quotes, no spaces, all toml, we failed at this many times i know from experience and i tested on a toml validator online.

Reply:
Zharinov why do u care about the capturing groups, im looking just for the "index" of the first match it doesn't mean anything in this case.
u mean u want it like this? `(?:id\\s*=\\s*)?['"]?${eDn}["']?(?:(?:\\s*=\\s*)|:|,\\s*)(?:.*version(?:\\.ref)?(?:\\s*\\=\\s*))?["']?${eVer}['"]?`

i dont think this is necessary though

Reply:
or is there a way to get the Index of a Specific group so i can make code easier?

Reply:
i tried to research that, you can't get the index of a specific group in the regex need to match the whole thing.. so its not necessary, do you still think i should add those extra non-capture groups and complicate the regex even more? XD

Reply:
use named capture groups for the parts you're interested in.

regex101.com is a good helper to test out.

Reply:
Thanks viceice/zharinov for teaching me new stuff about regex i never thought i'd use, 
after testing your theory, 
![image](https://user-images.githubusercontent.com/42116482/166141930-a24fc0ba-88ee-4d95-b864-4affdd1c8194.png)
There's no way to get the Index of an Individual group, i researched google too.

I don't think i need to change anything in here, since all i need is the index.

Reply:
i might be able to if i use this: https://stackoverflow.com/questions/15934353/get-index-of-each-capture-in-a-javascript-regex

String.prototype.matchAll but then i wouldn;t be using our `regEx`

Reply:
```suggestion
    `(?:id\\s*=\\s*)?['"]?${eDn}["']?(?:(?:\\s*=\\s*)|:|,\\s*)(?:.*version(?:\\.ref)?(?:\\s*\\=\\s*))?["']?${eVer}['"]?`
```

@PhilipAbed this is what @zharinov means, you'll still get your index for full match, but regex engine won't track individual match groups. So `match.length` will be `1`